File: 7/7.py
class Solution:
    def reverse(self, x: int) -> int:
        ans = int(str(abs(x))[::-1]) * (1 if x>=0 else -1) 
        if ans < -2**31 or ans > 2**31-1:
            return 0
        else:
            return  ans
        

File: 7/7.cpp
#include <algorithm>
#include <iostream>
#include <iterator>
#include <string>

using namespace std;

class Solution {
public:
	int reverse(int x) {
		string rev = to_string(x);
		auto start_iter = rev.begin() + (x < 0 ? 1 : 0);
		std::reverse(start_iter, rev.end());
		long long ret = std::stoll(rev);
		ret = (ret > 2147483647 or ret < -2147483648) ? 0 : ret;
		return ret;
	}
};
